src/lib/util/domains/domains.ts

```diff
@@ -52,3 +52,18 @@ export const getEtherscanLink = (domain: Domain, chainId: Network) =>
 
 		return `${etherscanBaseUri}token/${registrarAddress}?a=${domainIdInteger?.toString()}`;
 	}, [domain, domain?.id, chainId]);
+
+// TODO: move to utils
+export const truncateDomain = (
+	domainName: string,
+	maxCharacterLength: number,
+) => {
+	if (domainName?.length > maxCharacterLength) {
+		const splits = domainName.split('.');
+		if (splits?.length > 2) {
+			return `${splits[0]}...${splits[splits?.length - 1]}`;
+		}
+		return domainName;
+	}
+	return domainName;
+};
```

Comment:
This one is in `zapp-utils`.

Reply:
We have `truncateAddress` in `zapp-utils`. The aim of the util above is to format the `zna` to something like `wilder...berry`.

PR added: https://github.com/zer0-os/zApp-utils/pull/1

Reply:
Ah yes I see, makes sense!